### PR TITLE
Fix crashing when a spy's "real" owner gets changed

### LINKED ISSUE #11996: Evacuating spies in Exodus mission crashes the game
```
Red Alert Mod at Version playtest-20160904 
on map 492158a8d258b6aa7220a117b2ac2ac653151a46 (Exodus by Scott_NZ). 
Date: 2016-09-06 21:36:03Z 
Operating System: Windows (Microsoft Windows NT 6.1.7601 Service Pack 1) 
Runtime Version: .NET CLR 4.0.30319.42000 
Exception of type `System.NullReferenceException`: Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt. 
bei OpenRA.Mods.Common.Traits.Render.RenderDebugState.GetColor() 
bei OpenRA.Mods.Common.Traits.Render.RenderDebugState.OpenRA.Traits.INotifyOwnerChanged.OnOwnerChanged(Actor self, Player oldOwner, Player newOwner) 
bei OpenRA.Actor.<ChangeOwner>c__AnonStorey3.<>m__0(World w) 
bei OpenRA.World.Tick() 
bei OpenRA.Game.InnerLogicTick(OrderManager orderManager) 
bei OpenRA.Game.LogicTick() 
bei OpenRA.Game.Loop() 
bei OpenRA.Game.Run() 
bei OpenRA.Program.Run(String[] args) 
bei OpenRA.Program.Main(String[] args) 
```

http://www.sleipnirstuff.com/forum/viewtopic.php?f=82&t=19774

### PR DESCRIPTION
**Pull request by obrakmann**
Fixes #11996

### CODE REVIEW

**Review by reaperrr — APPROVED** [review]

**Comment by Mailaender** [comment]
Works as promised. :+1: You can also remove my workaround now:

```diff
diff --git a/mods/ra/maps/infiltration/rules.yaml b/mods/ra/maps/infiltration/rules.yaml
index 47b0263..ec2da22 100644
--- a/mods/ra/maps/infiltration/rules.yaml
+++ b/mods/ra/maps/infiltration/rules.yaml
@@ -54,9 +54,6 @@ SPY.Strong:
                Types: MissionObjective
        ExternalCaptures:
                CaptureTypes: MissionObjective
-       EditorOnlyTooltip:
-               Description: RenderDebugState.OnOwnerChange will crash
-       -RenderDebugState:
        Passenger:
                GrantUpgrades: mobile
```

**Comment by obrakmann** [comment]
Removed workarounds.

File: OpenRA.Mods.Common/Traits/Render/RenderDebugState.cs [hunk 1]
```diff
@@ -65,7 +65,7 @@ void INotifyOwnerChanged.OnOwnerChanged(Actor self, Player oldOwner, Player newO
 
 		Color GetColor()
 		{
-			return self.EffectiveOwner != null ? self.EffectiveOwner.Owner.Color.RGB : self.Owner.Color.RGB;
+			return self.EffectiveOwner != null && self.EffectiveOwner.Owner != null ? self.EffectiveOwner.Owner.Color.RGB : self.Owner.Color.RGB;
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Render/RenderDebugState.cs]: This should be looking at `EffectiveOwner.Disguised` for consistency with the other checks.

**Review by pchote — CHANGES_REQUESTED** [review]

**Comment by obrakmann** [comment]
Changed

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/6c079644da99005c086f8ce266ed7f89d549978f

**Pull request merged**